tests/TestCase/Database/Type/DecimalTypeTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Database\Type;
+
+use Cake\Database\Type;
+use Cake\Database\Type\DecimalType;
+use Cake\I18n\I18n;
+use Cake\TestSuite\TestCase;
+use \PDO;
+
+/**
+ * Test for the Decimal type.
+ */
+class DecimalTypeTest extends TestCase
+{
+
+    /**
+     * Setup
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+        $this->type = Type::build('decimal');
+        $this->driver = $this->getMockBuilder('Cake\Database\Driver')->getMock();
+        $this->locale = I18n::locale();
+        $this->numberClass = DecimalType::$numberClass;
+
+        I18n::locale($this->locale);
+    }
+
+    /**
+     * tearDown method
+     *
+     * @return void
+     */
+    public function tearDown()
+    {
+        parent::tearDown();
+        I18n::locale($this->locale);
+        DecimalType::$numberClass = $this->numberClass;
+    }
+
+    /**
+     * Test toPHP
+     *
+     * @return void
+     */
+    public function testToPHP()
+    {
+        $this->assertNull($this->type->toPHP(null, $this->driver));
+
+        $result = $this->type->toPHP('some data', $this->driver);
+        $this->assertSame(0.0, $result);
+
+        $result = $this->type->toPHP('2', $this->driver);
+        $this->assertSame(2.0, $result);
+
+        $result = $this->type->toPHP('2 bears', $this->driver);
+        $this->assertSame(2.0, $result);
```

Comment:
Sure but this is an abstraction on top of PHP that can be more strict than vanilla PHP. It is a Type class after all?